git_pile/config.py

```diff
@@ -36,6 +44,17 @@ def per_worktree(cls):
         return git_can_fail("config --get --bool extensions.worktreeConfig", stderr=nul_f).stdout.strip() == "true"
 
     def __init__(self):
+        if Config.per_worktree():
+            self.write = run_wrapper(["git", "config", "--worktree"], capture=True)
+        else:
+            self.write = run_wrapper(["git", "config"], capture=True)
+
+        # assume the call is from main worktree, where the result-branch is
+        # normally checkout
+        self._load_config_values(".")
```

Comment:
Hm... Thinking again about this one: Shouldn't we be using the output of `git rev-parse --show-toplevel` here? Since we are using that for the `root` attribute...

Reply:
Yeah... if we don't do this, we may fail on commands that don't go over normalize():  init and setup

Reply:
I was going to used it, but noticed we have a circular dep between config.py and gitutils.py and I didn't won't to solve it now. So I ended up trying this shortcut, but yes... it's not the appropriate solution